system/datatypes/peer_reviewed_article.php

```diff
@@ -1462,6 +1463,17 @@ protected function createCommentDomElement(HyphaDomElement $comment, $review = f
 			return '<span><a href="' . $url . '" target="_blank">' . $url . '</a>' . $suffix.'</span>';
 		}, $commentHtml);
 		$commentHtml = nl2br($commentHtml);
+		if(strlen($commentHtml) > self::MAX_COMMENT_LENGTH){
+			$id = rand();
+			$stub = substr(strip_tags($commentHtml),0, self::MAX_COMMENT_LENGTH*0.7);
+			$commentHtml = '<div style="display:none;" id="commentFull'.$id.'">'.$commentHtml.'</div>
+			<span id="commentStub'.$id.'">'.$stub.'...</span>
```

Comment:
Why use `$commentHtml` here and strip_tags that? That effectively just cancels the link-detection and nl2br?

Also even without HTML tags, there's currently a chance of cutting halfway an entity (i.e. `&amp;`) which could break the HTML (or at least result in an unsightly `&amp` or so being shown).

Wouldn't it be better to cut the original unprocessed comment and then apply all processing (htmlspecialchars, link detection, nl2br) all on the stub? AFAICS that should result in consistent and safe output, that has newlines in all the places you'd expect. The only caveat I can think of is that links might be cut halfway, which could be fixed by skipping the linkifying for a match at the very end of the comment, or by cutting on spaces rather than halfway words too.

Of course this does require that you remember that a comment was cut so you can add the "read more" link *after* all the processing, rather than before.

Reply:
Changed the process according to your feedback :)

Reply:
Looks great, thanks!